src/main/kotlin/org/rust/lang/core/types/infer/Expressions.kt

```diff
@@ -127,10 +127,24 @@ fun inferExpressionType(expr: RsExpr): RustType {
                 RustUnknownType
         }
 
+        is RsArrayExpr -> inferArrayType(expr)
+
         else -> RustUnknownType
     }
 }
 
+private fun inferArrayType(expr: RsArrayExpr): RustType {
+    val (elementType, size) = if (expr.semicolon != null) {
+        if (expr.exprList.size != 2) return RustUnknownType
+        val (elementExpr, sizeExpr) = expr.exprList
+        val size = calculateArraySize(sizeExpr) ?: return RustUnknownType
```

Comment:
But `RsArrayType` requires not only element type but size too. What is it supposed to be array size when we can't calculate it?